src/zmq/zmqpublishnotifier.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "zmqpublishnotifier.h"
+#include "main.h"
+#include "net.h"
```

Comment:
Whenever you use something defined by a header H in X.c, H needs to be
#included by either X.c or X.h.

You should not remove #includes because they're already included by one of
the dependencies. That leads to unclear dependency graphs, and may cause
breakage when dependencies change.